src/graph_utils.rs

```diff
@@ -64,10 +70,61 @@ pub fn find_bridges(nodes: &[Node], threshold_adjustment: f64) -> HashMap<usize,
     bridges
 }
 
+pub fn construct_graph(nodes: &[Node]) -> Graph<IpAddr> {
+    let mut graph = Graph::new();
+
+    for i in 0..nodes.len() {
```

Comment:
`iter().enumarate()` could be a more Rusty solution maybe

Reply:
Done.